kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -324,6 +357,86 @@ void shouldIncrementCountOnMessageTransit(UnaryOperator<ConfigurationBuilder> bu
+                                .singleElement()
+                                .value()
+                                .isGreaterThanOrEqualTo(1.0)),
+                argumentSet("counts connections to node, client to proxy",
+                        (UnaryOperator<ConfigurationBuilder>) builder -> builder,
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_client_to_proxy_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .isEmpty(),
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_client_to_proxy_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .singleElement()
+                                .value()
+                                .isGreaterThanOrEqualTo(1.0)),
+                argumentSet("count bootstrap connection from proxy to server ",
+                        (UnaryOperator<ConfigurationBuilder>) builder -> builder,
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "bootstrap")
+                                .isEmpty(),
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "bootstrap")
+                                .singleElement()
+                                .value()
+                                .isGreaterThanOrEqualTo(1.0)),
+                argumentSet("count node connection from proxy to server ",
+                        (UnaryOperator<ConfigurationBuilder>) builder -> builder,
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .isEmpty(),
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .singleElement()
+                                .value()
+                                .isGreaterThanOrEqualTo(1.0)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("connectionMetricsScenarios")
+    void shouldIncrementCountOnConnectionMetrics(UnaryOperator<ConfigurationBuilder> builder,
```

Comment:
the builder argument is redundant in your case.  In transitScenarios I'm using it adjust the filter configuration so I can control whether the proxy decodes the message or not.   As shouldIncrementCountOnConnectionMetrics is test happy path, you don't need it.